tt_metal/detail/tt_metal.hpp

```diff
@@ -27,7 +27,20 @@ namespace tt::tt_metal{
 
     namespace detail {
 
+        inline bool bypassDispatchCheck = false;
+
+        inline static void BypassDispatchCheck (){
```

Comment:
Instead of adding a bypass dispatch check here, can you try using the `fd_bootloader_mode` flag in `ConfigureDeviceWithProgram`? You can propagate it to `LaunchProgram` if needed.

Reply:
I see, basically add a force slow launch flag to `LaunchProgram`. Yeah that should be less code.

Reply:
hm Im not sure how I feel about an additional parameter in LaunchProgram since its user facing, is there another way that we can check whether a program is part of boot loading?

Reply:
I originally had this 
https://github.com/tenstorrent/tt-metal/commit/6e57f3b22b626a2c448d9da61a6af5d41ac039cf
What do you think about this one?

Reply:
yeah I think that makes sense. I guess LaunchProgram is in detail namespace as well so maybe its okay to add params there? what do you prefer @aliuTT

Reply:
@abhullar-tt I preferred adding a param to LaunchProgram over adding more detail:: apis to bypass the dispatch check. I remember we were saying that the dispatch state check could be removed eventually, so I thought it would also be easier to remove it from LaunchProgram. Overall, no strong preference either way... I think my favorite approach would have been to use ConfigureDeviceWithProgram, and sending the launch msgs manually, sort of like how we did it for dispatch kernels.

Reply:
hm yeah adding the extra param  makes sense to me!